typescript/web/src/pages/[workspaceSlug]/datasets/index.tsx

```diff
@@ -41,6 +54,62 @@ export const getDatasetsQuery = gql`
   }
 `;
 
+const MigrateLocalDatasetsModal = () => {
+  const router = useRouter();
+  const workspaceSlug = router?.query.workspaceSlug as string;
+  const [isLocalDatasetsModalOpen, setIsLocalDatasetsModalOpen] =
+    useState<boolean>(false);
+  const closeModal = useCallback(() => {
+    setIsLocalDatasetsModalOpen(false);
+  }, []);
+  useEffect(() => {
+    if (workspaceSlug === "local") {
+      setIsLocalDatasetsModalOpen(true);
+    }
+  }, [workspaceSlug]);
+  return (
+    <Modal
+      isOpen={isLocalDatasetsModalOpen}
+      onClose={closeModal}
+      isCentered
+      size="xl"
+    >
+      <ModalOverlay />
+      <ModalContent>
+        <ModalCloseButton />
+        <ModalHeader>
+          <Heading as="h2" size="md">
```

Comment:
I wonder why you set both `h2` and the `size` 🤔

Reply:
`as` is more a semantically prop to make appear a `<h2>...</h2>` in the DOM whereas the `size` is really to control how big/small it will visually appear to the user.

**However**, `Heading` render `h2` by default so I will remove it 😛